ipa-core/tests/hybrid.rs

```diff
@@ -134,3 +146,213 @@ fn test_hybrid() {
         .zip(expected_result.iter())
         .all(|(a, b)| a == b));
 }
+
+#[test]
+fn test_hybrid_poll() {
+    const INPUT_SIZE: usize = 100;
+    const SHARDS: usize = 5;
+    const MAX_CONVERSION_VALUE: usize = 5;
+
+    let config = HybridQueryParams {
+        max_breakdown_key: 5,
+        with_dp: 0,
+        epsilon: 0.0,
+        // only encrypted inputs are supported
+        plaintext_match_keys: false,
+    };
+
+    let dir = TempDir::new_delete_on_drop();
+
+    // Gen inputs
+    let input_file = dir.path().join("ipa_inputs.txt");
+    let in_the_clear_output_file = dir.path().join("ipa_output_in_the_clear.json");
+    let output_file = dir.path().join("ipa_output.json");
+
+    let mut command = Command::new(TEST_RC_BIN);
+    command
+        .args(["--output-file".as_ref(), input_file.as_os_str()])
+        .arg("gen-hybrid-inputs")
+        .args(["--count", &INPUT_SIZE.to_string()])
+        .args(["--max-conversion-value", &MAX_CONVERSION_VALUE.to_string()])
+        .args(["--max-breakdown-key", &config.max_breakdown_key.to_string()])
+        .args(["--seed", &thread_rng().next_u64().to_string()])
+        .silent()
+        .stdin(Stdio::piped());
+    command.status().unwrap_status();
+
+    let mut command = Command::new(IN_THE_CLEAR_BIN);
+    command
+        .args(["--input-file".as_ref(), input_file.as_os_str()])
+        .args([
+            "--output-file".as_ref(),
+            in_the_clear_output_file.as_os_str(),
+        ])
+        .silent()
+        .stdin(Stdio::piped());
+    command.status().unwrap_status();
+
+    let config_path = dir.path().join("config");
+    let sockets = test_sharded_setup::<SHARDS>(&config_path);
+    let _helpers = spawn_shards(&config_path, &sockets, true);
+
+    // encrypt input
+    let mut command = Command::new(CRYPTO_UTIL_BIN);
+    command
+        .arg("hybrid-encrypt")
+        .args(["--input-file".as_ref(), input_file.as_os_str()])
+        .args(["--output-dir".as_ref(), dir.path().as_os_str()])
+        .args(["--length-delimited"])
+        .args(["--network".into(), config_path.join("network.toml")])
+        .stdin(Stdio::piped());
+    command.status().unwrap_status();
+    let enc1 = dir.path().join("helper1.enc");
+    let enc2 = dir.path().join("helper2.enc");
+    let enc3 = dir.path().join("helper3.enc");
+
+    let poll_port = TcpListener::bind("127.0.0.1:0").unwrap();
```

Comment:
should this be some random high value port?

Reply:
That's what `:0` will do.

Reply:
til